transport/src/test/java/io/zeebe/transport/impl/AtomixTransportTest.java

```diff
@@ -260,17 +265,18 @@ public void shouldRetryAndSucceedAfterNodeIsResolved() throws InterruptedExcepti
     final var nodeAddressRef = new AtomicReference<String>();
     final var retryLatch = new CountDownLatch(3);
     serverTransport.subscribe(0, new DirectlyResponder()).join();
+
     final var requestFuture =
         clientTransport.sendRequestWithRetry(
             () -> {
               retryLatch.countDown();
               return nodeAddressRef.get();
             },
             new Request("messageABC"),
-            Duration.ofSeconds(5));
```

Comment:
You change the behavior of some tests, might make the tests more flaky, since now you have the request timeout and latch timeout as same value.

Reply:
I don't think that I changed the behavior of the tests :thinking: Setting the request timeout to the same as the timeout of the latch should be ok because the request and latch should be finished before 10 seconds.

Please correct me if I'm wrong.

Reply:
This is just an assumption, right? Since we don't know why we have the execution pauses it can also stop longer right? If the latch and timeout is the same it can happen that the timeout happens before. We not asserting that the latch succeed btw. before it waited until it the latch was triggered now you added a timeout which means you need to assert it.

Reply:
I reduced the timeout of the latch. If the request takes longer then the test has a higher chance to pass.

I avoid checking the latch result for more stability. Otherwise, a test would fail if it retries just 2 of 3 times. The actual behavior is tested afterward (i.e. checking the response future).